packages/origin-ui-core/src/components/bundles/BundleItemEdit.tsx

```diff
@@ -0,0 +1,231 @@
+import React, { useState, useEffect } from 'react';
+import {
+    Box,
+    useTheme,
+    Grid,
+    ListItemAvatar,
+    Avatar,
+    Button,
+    IconButton,
+    FormControl,
+    InputLabel,
+    Input,
+    InputAdornment,
+    makeStyles
+} from '@material-ui/core';
+import { ICertificateViewItem } from '../../features/certificates';
+import { ICertificateEnergy } from '../../../../issuer/src';
+import { BigNumber } from 'ethers';
+import {
+    energyImageByType,
+    moment,
+    EnergyFormatter,
+    deviceById,
+    getEnvironment,
+    getProducingDevices,
+    EnergyTypes
+} from '../..';
+import { useSelector } from 'react-redux';
+import { useTranslation, useValidation } from '../../utils';
+import { Edit, HighlightOff } from '@material-ui/icons';
+import { Formik, Field, Form } from 'formik';
+import variables from '../../styles/variables.scss';
+
+export interface IBundledCertificateEnergy extends ICertificateEnergy {
+    volumeToBundle: BigNumber;
+}
+
+export interface IBundledCertificate extends ICertificateViewItem {
+    energy: IBundledCertificateEnergy;
+}
+
+interface IOwnProps {
+    certificate: IBundledCertificate;
+    totalVolume: () => BigNumber;
+    onChange: (cert: IBundledCertificate) => void;
+}
+
+export const BundleItemEdit = (props: IOwnProps) => {
+    const { t } = useTranslation();
+    const { onChange } = props;
+    let totalVolume = props.totalVolume;
+    const { certificate } = props;
+    const {
+        creationTime,
+        deviceId,
+        energy: { publicVolume, volumeToBundle }
+    } = certificate;
+    const [selected, setSelected] = useState<boolean>(false);
+    const {
+        typography: { fontSizeMd },
+        spacing
+    } = useTheme();
+    const environment = useSelector(getEnvironment);
+    const devices = useSelector(getProducingDevices);
+    const classes = makeStyles(() => ({
+        formControl: {
+            width: '100%',
+            backgroundColor: variables.backgroundColorDarker
+        },
+        formLabel: {
+            marginTop: spacing(1),
+            marginLeft: spacing(1)
+        },
+        formInput: {
+            textAlign: 'right'
+        }
+    }))();
+
+    useEffect(() => {
+        totalVolume = props.totalVolume;
+    }, [props.totalVolume]);
+
+    const { province, deviceType, facilityName } = deviceById(deviceId, environment, devices);
+    const type = deviceType.split(';')[0].toLowerCase() as EnergyTypes;
+
+    const { Yup } = useValidation();
+    const validationSchema = Yup.object().shape({
+        volumeToBundle: Yup.number()
+            .required()
+            .min(0)
+            .max(Number(EnergyFormatter.format(publicVolume)))
```

Comment:
Cleaner way of getting the number is `.max(publicVolume.toNumber())` (instead of coverting the number to string and then casting to Number).

Reply:
But we still need `EnergyFormatter.format` which converts Wh to MWh and returns string

Reply:
For converting between MWh and Wh please use either `EnergyFormatter.getValueInDisplayUnit()` or `EnergyFormatter.getBaseValueFromValueInDisplayUnit()`.

`EnergyFormatter.format()` should only be used to print, not to convert.